src/app/views/editors/ckeditor/ckeditor_upload_plugin.ts

```diff
@@ -1,26 +1,75 @@
+///<reference path="../upload/renuo_upload_html_generator.ts"/>
+///<reference path="../upload/renuo_upload.d.ts"/>
 class CkeditorUploadPlugin {
+  constructor(private domContentBlock:DomContentBlock) {
+  }
+
   registerPlugin(editor:CKEDITOR.editor) {
-    return;
-
-    // TODO: implement this
-    //editor.addCommand('renuoUpload', {
-    //  exec: function (editor) {
-    //    const now = new Date();
-    //    editor.insertHtml('The current date and time is: <em>' + now.toString() + '</em>');
-    //  }
-    //});
-
-    //const pluginName = 'renuoUpload';
-    //CKEDITOR.plugins.add(pluginName, {
-    //  init: function (editor) {
-    //    editor.addCommand(pluginName, startUpload);
-    //    editor.ui.addButton(pluginName, {
-    //      label: 'Renuo Upload',
-    //      // icon: 'renuo-upload.gif',
-    //      toolbar: 'insert',
-    //      command: pluginName
-    //    });
-    //  }
-    //});
+    if (!this.domContentBlock.hasRenuoUpload()) return true;
+
+    editor.addCommand('renuoUpload', {
+      exec: (editor) => {
+        this.openDropzone(editor);
+        return true;
+      }
+    });
+    editor.ui.addButton('uploadButton', {
+      label: 'File/Image Upload',
+      command: 'renuoUpload',
+      toolbar: 'insert',
+      icon: 'imagebutton'
+    });
   }
+
+  private closeDropzone():void {
+    jQuery('#dropzone-overlay').remove();
+  }
+
+  private openDropzone(editor:CKEDITOR.editor):void {
```

Comment:
this whole method is quite long. can we refactor it? e.g. i would take the whole html building blocks out, as well a the css code.